lib/Conversion/LWEToPolynomial/LWEToPolynomial.cpp

```diff
@@ -116,6 +124,176 @@ struct ConvertRLWEDecrypt : public OpConversionPattern<RLWEDecryptOp> {
+      // multiplication.
+      auto constantT = builder.create<arith::ConstantOp>(
+          builder.getI32IntegerAttr(1 << cleartextBitwidth));
+
+      // TODO: Add pass options to change the seed (which is currently hardcoded
+      // to 0).
+
+      // Initialize random number generator with seed.
+      auto generateRandom =
+          builder.create<random::InitOp>(index0, builder.getI32IntegerAttr(32));
+
+      // Create a uniform discrete random distribution with generated values of
+      // -1, 0, 1.
+      auto uniformDistributionType = random::DistributionType::get(
+          getContext(), random::Distribution::uniform);
+
+      auto uniformDistribution =
+          builder.create<random::DiscreteUniformDistributionOp>(
+              uniformDistributionType, generateRandom,
+              builder.getI32IntegerAttr(-1), builder.getI32IntegerAttr(2));
+
+      // generate random u polynomial from uniform random ternary distribution
+      auto uTensor =
+          builder.create<random::SampleOp>(tensorParams, uniformDistribution);
+      auto u =
+          builder.create<polynomial::FromTensorOp>(uTensor, inputT.getRing());
+
+      // Create a discrete Gaussian distribution
+      auto discreteGaussianDistributionType = random::DistributionType::get(
+          getContext(), random::Distribution::gaussian);
+
+      auto discreteGaussianDistribution =
+          builder.create<random::DiscreteGaussianDistributionOp>(
+              discreteGaussianDistributionType, generateRandom,
+              builder.getI32IntegerAttr(0),
+              builder.getI32IntegerAttr(
+                  5));  // TODO: Add pass options to configure stdev (which is
+                        // currently hardcoded to 5)
+
+      // generate random e0 polynomial from discrete gaussian distribution
+      auto e0Tensor = builder.create<random::SampleOp>(
+          tensorParams, discreteGaussianDistribution);
+      auto e0 =
+          builder.create<polynomial::FromTensorOp>(e0Tensor, inputT.getRing());
+
+      // generate random e1 polynomial from discrete gaussian distribution
+      auto e1Tensor = builder.create<random::SampleOp>(
+          tensorParams, discreteGaussianDistribution);
+      auto e1 =
+          builder.create<polynomial::FromTensorOp>(e1Tensor, inputT.getRing());
+
+      // Computing ciphertext0 = publicKey0 * u + e0 * constantT + input
+      auto publicKey0U = builder.create<polynomial::MulOp>(publicKey0, u);
+      auto tE0 = builder.create<polynomial::MulScalarOp>(e0, constantT);
+      auto pK0UtE0 = builder.create<polynomial::AddOp>(publicKey0U, tE0);
+      auto ciphertext0 = builder.create<polynomial::AddOp>(pK0UtE0, input);
+
+      // Computing ciphertext1 = publicKey1 * u + e1 * constantT
+      auto publicKey1U = builder.create<polynomial::MulOp>(publicKey1, u);
+      auto tE1 = builder.create<polynomial::MulScalarOp>(e1, constantT);
+      auto ciphertext1 = builder.create<polynomial::AddOp>(publicKey1U, tE1);
```

Comment:
what ends up getting multiplied by what constant is different across different schemes

Reply:
So far, the lowering only supports BGV. Is the constant multiplication and the cleartext bitwidth / cleartext start the only thing that changes across different schemes? What can we be sure of generalizing across the different RLWE schemes that we want to support? We can either leave this generalization as a future PR or if it's simple enough, I can try to generalize it.

Reply:
I think there'll need to be two places where the pass effectively switch-cases on (a) the plaintext encoding attribute and (b) the encryption info attribute (which is the thing with multiplying the noise vs multiplying the message). But it should be sufficient to add TODOs for that and otherwise merge as-is

Reply:
Yep, please just add a TODO - I expect we'll bikeshed a bit on https://github.com/google/heir/pull/876 and when I do the migration on the types I will just support the MSB encoding for BG/FV and someone can add the CKKS style one later.